### PR TITLE
refactor(client): decompose order page and bug fixes

### PR DESCRIPTION
**Pull request by gordonpn**
#

## Changes this PR introduces

- Fix bug where every first order would fail because the states (order date and delivery date) were not ready (they are asynchronous) for the post request
- Fix bug where if user did not change quantity field, the default 0 would be used thus failing the post request
- Fix bug where if you submit a successful order, you can submit another successful one without even selecting the material because the state was not reset
- Extract chunks of logic out of functional components

## How to test this PR

- Visually it should still be the same

## ClickUp reference 

CU-q38gfh

## Notes for reviewers

- Feel free to ask me anything
- Disregard Zeet failing
- Report any bugs you see, I will fix them

## Checklist

- [x] Have I renamed the title of this pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)?
- [x] I will squash and merge when my PR is approved.
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-q38gfh Refactor Order page to use a custom hook](https://app.clickup.com/t/q38gfh)

**Comment by gordonpn** [comment]
Disregard fucking zeet

**Review by tiffzeng — APPROVED** [review]

**Review by Mackbellemore — APPROVED** [review]

File: client/src/components/Order/FormModal.jsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  Select,
+  Textarea,
+} from '@chakra-ui/react';
+import useOrderForm from 'hooks/useOrderForm.jsx';
+import PropTypes from 'prop-types';
+import React, { Fragment } from 'react';
+
+const FormModal = ({ showButton = false }) => {
+  // TODO: use the endpoint to get this constant list of materials rather than hard coding
+  const materialTypes = ['rubber', 'aluminum', 'steel', 'copper', 'plastic', 'leather'];
```
Comment (DavidLiang01) [inline client/src/components/Order/FormModal.jsx:29]: is it better if we put `materialTypes ` in the constants.js file?

Comment (gordonpn) [reply client/src/components/Order/FormModal.jsx:29]: Yes, at the very least it should be in the `constants.js` but I left a TODO for Tiff to take a look at (same comment from her PR)

Comment (Mackbellemore) [reply client/src/components/Order/FormModal.jsx:29]: i believe naasir is adding code to fetch this from the backend, so we could just wait for that

**Review by DavidLiang01 — APPROVED** [review]

**Pull request merged**